version: '3.8'
services:
  db:
    image: postgres:15
    restart: always
    ports:
      - '5432:5432'
    volumes:
      - ./db/init.sql:/docker-entrypoint-initdb.d/init.sql
    env_file:
      - ../.env

  lucidify:
    image: lucidify
    build:
      context: ../frontend/lucidify
      dockerfile:  Dockerfile.dev
    ports:
      - '3001:3000'
    volumes:
      - ../frontend/lucidify:/app
    env_file:
      - ../.env

  lucidify-api:
    image: backend/lucidify-api 
    build:
      context: ../backend/lucidify-api
      dockerfile:  Dockerfile.dev
    ports:
      - '8080:8080'
    volumes:
      - ../backend/lucidify-api:/app
    env_file:
      - ../.env

  weaviate:
    image: semitechnologies/weaviate:1.21.3
    ports:
      - 8090:8080
    restart: on-failure:0
    environment:
      QUERY_DEFAULTS_LIMIT: 25
      AUTHENTICATION_ANONYMOUS_ACCESS_ENABLED: 'true'
      PERSISTENCE_DATA_PATH: '/var/lib/weaviate'
      DEFAULT_VECTORIZER_MODULE: 'text2vec-openai'
      ENABLE_MODULES: 'text2vec-openai,generative-openai'
      CLUSTER_HOSTNAME: 'node1'
    env_file:
      - ../.env

volumes:
  db:
    driver: local
  weaviate_data:
